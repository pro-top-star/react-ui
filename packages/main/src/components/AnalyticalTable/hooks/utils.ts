export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
