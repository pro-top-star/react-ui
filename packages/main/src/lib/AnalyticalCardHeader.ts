import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
