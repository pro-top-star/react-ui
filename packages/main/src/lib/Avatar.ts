import { Avatar } from '../webComponents/Avatar';

export { Avatar };
