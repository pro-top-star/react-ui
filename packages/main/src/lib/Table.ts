import { Table } from '../webComponents/Table';
import type { TablePropTypes } from '../webComponents/Table';

export { Table };
export type { TablePropTypes };
