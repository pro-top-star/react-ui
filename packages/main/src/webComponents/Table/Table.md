[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/Table)

```jsx
import { Table } from '@ui5/webcomponents-react/lib/Table';
```
