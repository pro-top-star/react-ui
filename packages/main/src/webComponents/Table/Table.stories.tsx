import { action } from '@storybook/addon-actions';
import { boolean, text } from '@storybook/addon-knobs';
import { Label } from '@ui5/webcomponents-react/lib/Label';
import { Table } from '@ui5/webcomponents-react/lib/Table';
import { TableCell } from '@ui5/webcomponents-react/lib/TableCell';
import { TableColumn } from '@ui5/webcomponents-react/lib/TableColumn';
import { TableRow } from '@ui5/webcomponents-react/lib/TableRow';
import React from 'react';

export default {
  title: 'UI5 Web Components / Table',
  component: Table,
  parameters: {
    subcomponents: { TableColumn, TableRow, TableCell }
  }
};

export const generatedDefaultStory = () => (
  <Table
    noDataText={text('noDataText', '')}
    showNoData={boolean('showNoData', false)}
    stickyColumnHeader={boolean('stickyColumnHeader', false)}
    columns={
      <>
        <TableColumn style={{ width: '12rem' }}>
          <Label>Product</Label>
        </TableColumn>
        <TableColumn minWidth={800} popinText="Supplier">
          <Label>Supplier</Label>
        </TableColumn>
        <TableColumn minWidth={600} popinText="Dimensions" demandPopin>
          <Label>Dimensions</Label>
        </TableColumn>
        <TableColumn minWidth={600} popinText="Weight" demandPopin>
          <Label>Weight</Label>
        </TableColumn>
        <TableColumn>
          <Label>Price</Label>
        </TableColumn>
      </>
    }
    onPopinChange={action('onPopinChange')}
    onRowClick={action('onRowClick')}
  >
    <TableRow>
      <TableCell>
        <Label>Notebook Basic</Label>
      </TableCell>
      <TableCell>
        <Label>Very Best Screens</Label>
      </TableCell>
      <TableCell>
        <Label>30 x 18 x 3cm</Label>
      </TableCell>
      <TableCell>
        <Label>4.2KG</Label>
      </TableCell>
      <TableCell>
        <Label>956EUR</Label>
      </TableCell>
    </TableRow>
    <TableRow>
      <TableCell>
        <Label>Notebook Basic 17HT-1001</Label>
      </TableCell>
      <TableCell>
        <Label>Very Best Screens</Label>
      </TableCell>
      <TableCell>
        <Label>29 x 17 x 3.1cm</Label>
      </TableCell>
      <TableCell>
        <Label>4.5KG</Label>
      </TableCell>
      <TableCell>
        <Label>1249EUR</Label>
      </TableCell>
    </TableRow>
  </Table>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
