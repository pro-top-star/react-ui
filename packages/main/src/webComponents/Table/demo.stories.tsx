import { boolean } from '@storybook/addon-knobs';
import { Label } from '@ui5/webcomponents-react/lib/Label';
import { Table } from '@ui5/webcomponents-react/lib/Table';
import { TableCell } from '@ui5/webcomponents-react/lib/TableCell';
import { TableColumn } from '@ui5/webcomponents-react/lib/TableColumn';
import { TableRow } from '@ui5/webcomponents-react/lib/TableRow';
import { Text } from '@ui5/webcomponents-react/lib/Text';
import React from 'react';

export default {
  title: 'UI5 Web Components / Table',
  component: Table,
  parameters: {
    subcomponents: { TableColumn, TableRow, TableCell }
  }
};

export const defaultStory = () => (
  <Table
    stickyColumnHeader={boolean('stickyColumnHeader', false)}
    columns={[
      <TableColumn>
        <Label>Column 1</Label>
      </TableColumn>,
      <TableColumn>
        <Text>Column 2</Text>
      </TableColumn>,
      <TableColumn>
        <Text>Column 3</Text>
      </TableColumn>
    ]}
  >
    <TableRow>
      <TableCell>
        <Label>Cell 1</Label>
      </TableCell>
      <TableCell>
        <Label>Cell 2</Label>
      </TableCell>
      <TableCell>
        <Label>Cell 3</Label>
      </TableCell>
    </TableRow>
    <TableRow>
      <TableCell>
        <Label>Cell 2.1</Label>
      </TableCell>
      <TableCell>
        <Label>Cell 2.2</Label>
      </TableCell>
      <TableCell>
        <Label>Cell 2.3</Label>
      </TableCell>
    </TableRow>
  </Table>
);

defaultStory.story = {
  name: 'Default'
};
