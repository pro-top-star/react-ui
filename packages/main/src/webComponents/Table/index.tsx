import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import '@ui5/webcomponents/dist/Table';
import React, { FC, ReactNode } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface TablePropTypes extends WithWebComponentPropTypes {
  /**
   * Defines the text that will be displayed when there is no data and <code>showNoData</code> is present.
   */
  noDataText?: string;
  /**
   * Defines if the value of <code>noDataText</code> will be diplayed when there is no rows present in the table.
   */
  showNoData?: boolean;
  /**
   * Determines whether the column headers remain fixed at the top of the page during vertical scrolling as long as the Web Component is in the viewport. <br><br> <b>Limitations:</b> <ul> <li>Browsers that do not support this feature: <ul> <li>Internet Explorer</li> <li>Microsoft Edge lower than version 41 (EdgeHTML 16)</li> <li>Mozilla Firefox lower than version 59</li> </ul> </li> <li>Scrolling behavior: <ul> <li>If the Web Component is placed in layout containers that have the <code>overflow: hidden</code> or <code>overflow: auto</code> style definition, this can prevent the sticky elements of the Web Component from becoming fixed at the top of the viewport.</li> </ul> </li> </ul>
   */
  stickyColumnHeader?: boolean;
  /**
   * Defines the configuration for the columns of the <code>ui5-table</code>. <br><br> <b>Note:</b> Use <code>ui5-table-column</code> for the intended design.
   */
  columns?: ReactNode | ReactNode[];
  /**
   * Defines the <code>ui5-table</code> rows. <br><br> <b>Note:</b> Use <code>ui5-table-row</code> for the intended design.
   */
  children?: ReactNode | ReactNode[];
  /**
   * Fired when the <code>ui5-table-column</code> is shown as a pop-in instead of hiding it.
   */
  onPopinChange?: (event: CustomEvent<{ poppedColumns: unknown[] }>) => void;
  /**
   * Fired when a row is clicked.
   */
  onRowClick?: (event: CustomEvent<{ row: ReactNode }>) => void;
}

/**
 * <code>import { Table } from '@ui5/webcomponents-react/lib/Table';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/Table" target="_blank">UI5 Web Components Playground</a>
 */
const Table: FC<TablePropTypes> = withWebComponent<TablePropTypes>(
  'ui5-table',
  ['noDataText'],
  ['showNoData', 'stickyColumnHeader'],
  ['columns'],
  ['popinChange', 'rowClick']
);

Table.displayName = 'Table';

Table.defaultProps = {
  showNoData: false,
  stickyColumnHeader: false
};

export { Table };
