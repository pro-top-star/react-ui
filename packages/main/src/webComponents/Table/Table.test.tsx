import { mountThemedComponent } from '@shared/tests/utils';
import { Table } from '@ui5/webcomponents-react/lib/Table';
import React from 'react';

describe('Table', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Table />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
