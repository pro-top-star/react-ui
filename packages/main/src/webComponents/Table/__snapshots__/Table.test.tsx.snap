// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Basic Test (generated) 1`] = `<ui5-table />`;
